### PR TITLE
[AERIE-1823]  sequence generation flow

### PR DESCRIPTION
**Pull request by dyst5422**
* **Tickets addressed:** AERIE-1823
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

Get the specified sequence + simulated activity ids + commands from the latest expansion run for each simulated activity (filtered on simulation dataset)
Get start time for each simulated activity and join with simulated activity command data
Call the seqbuilder on the sorted simulated activity to generate the sequence
Return the sequence as seqJson with sequence#toSeqJson

## Verification

Ran through testing locally.

## Documentation

[Create Sequence](https://github.com/NASA-AMMOS/aerie/wiki/Create-Sequence)
[Link Simulated Activity to Sequence](https://github.com/NASA-AMMOS/aerie/wiki/Link-Simulated-Activity-to-Sequence)
[Retrieve SeqJson Serialization of Sequence](https://github.com/NASA-AMMOS/aerie/wiki/Retrieve-SeqJson-Serialization-of-Sequence)

## Future work

Before merging:
- [x] test simple concat
- [x] test with non-existent expansion
- [x] test with expansions throwing errors

### CODE REVIEW

**Comment by dyst5422** [comment]
Apparently can't convert to draft, so applied the don't merge tag instead

File: command-expansion-server/src/app.ts
Comment (camargo) [inline command-expansion-server/src/app.ts]: Not necessarily this PR but the formatting for these queries is inconsistent from endpoint to endpoint in this file which might make longterm maintenance harder. Can we switch to use a consistent format? (caps vs lowercase, indentation, etc.)

Comment (dyst5422) [reply command-expansion-server/src/app.ts]: I was thinking about that the whole time I was working on this and had to resist the temptation for refactoring scope creep. When we get the rest of the PR finished up, I'll do a refactoring pass. + commit

Comment (goetzrrGit) [inline command-expansion-server/src/app.ts]: Are we using the expansion errors for anything? I can see adding the generated error message into the seqjson, but maybe that is a to-do for later as I am not sure where we would stick those errors in the seqJSON besides a `description` field.

File: command-expansion-server/src/app.ts
Comment (goetzrrGit) [inline command-expansion-server/src/app.ts:509]: Good idea to make this configurable

**Review by goetzrrGit — COMMENTED** [review]
Amazing work Dylan!!! I still need to test it though. I just had some generic questions.

File: command-expansion-server/src/app.ts
Comment (dyst5422) [reply command-expansion-server/src/app.ts]: I was just talking to matt about this. Can we put them into the seqJson simply under a non-existent command stem:
```
{
  stem: "$$ERROR$$",
  args: ["Error message here"],
  time: {
    tag: <activity start time>,
    type: "ABSOLUTE"
  },
  metadata: {}
}
```

File: command-expansion-server/src/app.ts
Comment (dyst5422) [reply command-expansion-server/src/app.ts:509]: Discussed with Pat and Matt and while this is useful, until there is a request for doing more complicated things when building the sequence, we won't be implementing this. And in fact, this might force that conversation of whether we should be doing sequence building at all and encourage them to just build a couple hundred line python script to pull from the api and do the building themselves.

File: command-expansion-server/src/app.ts
Comment (patkenneally) [inline command-expansion-server/src/app.ts]: :large_blue_circle: What are your thoughts on reformatting these long lines so they are initially visible without a wide monitor or an IDE auto-wrapping? 

Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: command-expansion-server/src/app.ts
Comment (patkenneally) [inline command-expansion-server/src/app.ts:396]: :large_blue_circle: Is this commit both fixing simulated activity batch loaders updated in the last commit and adding the SeqBuilder capability? I want to make sure I'm understanding the ideas/intentions introduced in this commit given the commit message.

File: command-expansion-server/src/app.ts
Comment (patkenneally) [inline command-expansion-server/src/app.ts:14]: :large_blue_circle: Seems like this commit has the message that should go with some of the code that enters the code base in [ff6fdf](https://github.com/NASA-AMMOS/aerie/pull/166/commits/ff6fdf9141556131dcc0b6081fb93b360dc171c5)

**Review by patkenneally — APPROVED** [review]

File: command-expansion-server/test/sequence-generation.spec.ts
Comment (mattdailis) [inline command-expansion-server/test/sequence-generation.spec.ts:1]: This test seems like it would do well as part of our `e2eTest` suite that runs on every PR.

**Review by mattdailis — APPROVED** [review]

File: command-expansion-server/src/app.ts
Comment (dyst5422) [reply command-expansion-server/src/app.ts:396]: Definitely a rebase error. I think all of this should be in 8128bba6d51d9e9b362a5b649c8bc99fefd241d8.

**Comment by dyst5422** [comment]
@mattdailis & @patkenneally sorry for the churn, but the easiest way to resolve the various rebasing issue was to just rewrite the whole commit history with a reset and cherry picking for a more cohesive commit story.

**Review by mattdailis — APPROVED** [review]
> @mattdailis & @patkenneally sorry for the churn, but the easiest way to resolve the various rebasing issue was to just rewrite the whole commit history with a reset and cherry picking for a more cohesive commit story.

No worries - I too got confused by the commits while reviewing, so I think the re-write was a good idea :+1:

File: command-expansion-server/src/app.ts
Comment (dyst5422) [reply command-expansion-server/src/app.ts]: Did a full auto-format with prettier which also addressed this

**Comment by dyst5422** [comment]
rebase on develop + fix up tests

File: command-expansion-server/test/sequence-generation.spec.ts
Comment (dyst5422) [reply command-expansion-server/test/sequence-generation.spec.ts:1]: Would love to add this in. Not sure how to and won't hold up this PR for it.

**Pull request merged**
